reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/collector/ir/LibraryQueriesResultsCollector.kt

```diff
@@ -22,10 +31,43 @@ class LibraryQueriesResultsCollector(
     override fun generate(moduleFragment: IrModuleFragment, pluginContext: IrPluginContext) {
         val irInstances = irInstancesAnalyzer.getIrInstances()
         val analyzer = IrReflektQueriesAnalyzer(irInstances, pluginContext)
+        val mentionedClasses: MutableSet<IrClass> = HashSet()
         messageCollector?.log("Start filtering instances for the ReflektImpl file")
         libraryQueriesResults.classes.filterInstances(analyzer, ReflektEntity.CLASSES)
+
+        mentionedClasses += libraryQueriesResults.classes.values.flatMap {
+            it + it.flatMap { cls -> cls.getAllSuperclasses() + cls.sealedSubclasses.map { subclass -> subclass.owner } }
+        }
+
         libraryQueriesResults.objects.filterInstances(analyzer, ReflektEntity.OBJECTS)
         libraryQueriesResults.functions.filterInstances(analyzer, ReflektEntity.FUNCTIONS)
+
+        libraryQueriesResults.mentionedClasses += mentionedClasses.map { cls ->
+            SerializableReflektClass(
```

Comment:
Maybe should we add a method for `SerializableReflektClass` that converts a class into `SerializableReflektClass`? The same thing is for `DescriptorVisibilities` vs `ReflektVisibility`

Reply:
OK.